totoify_grafeas/totoifylib.py

```diff
@@ -0,0 +1,277 @@
+    grafeas_occurrence.intoto['signatures'] = occ_json['intoto']['signatures']
+    grafeas_occurrence.resource['uri'] = occ_json['resource']['uri']
+    grafeas_occurrence.noteName = occ_json['noteName']
+    return grafeas_occurrence
+
+
+
+def create_grafeas_occurrence_from_in_toto_link(in_toto_link, step_name, resource_uri):
+  """Creates a GrafeasInTotoOccurrence class from an in-toto link Metablock class.
+
+  Arguments:
+    in_toto_link: A link of type Metablock (and we know that "signed" corresponds
+        to an object of type in_toto.models.link.Link)
+    step_name: A string of the step name
+    resource_uri: A string of the resource URI
+
+  Raises:
+    N/A
+
+  Side Effects:
+    N/A
+
+  Returns:
+    A GrafeasInTotoOccurrence object.
+  """
+  # TODO: set step_name from link if None
+  return GrafeasInTotoOccurrence(in_toto_link, step_name, resource_uri)
+
+
+
+def create_in_toto_link_from_grafeas_occurrence(grafeas_occurrence, step_name):
+  """Creates an in-toto link Metablock class from a GrafeasInTotoOccurrence class.
+
+  Arguments:
+    grafeas_occurrence: An occurrence of type GrafeasInTotoOccurrence.
+    step_name: A string of the step name.
+
+  Raises:
+    N/A
+
+  Side Effects:
+    N/A
+
+  Returns:
+    A Metablock object.
+  """
+  materials = {}
+  products = {}
+  command = []
+  byproducts = {}
+  environment = {}
+
+  for item in grafeas_occurrence.intoto["signed"].materials:
+    materials[item["resource_uri"]] = item["hashes"]
+
+  for item in grafeas_occurrence.intoto["signed"].products:
+    products[item["resource_uri"]] = item["hashes"]
+
+  command = grafeas_occurrence.intoto["signed"].command
+
+  for key, value in grafeas_occurrence.intoto["signed"].byproducts.items():
+    if key == "custom_values":
+      continue
+    byproducts[key] = value
+
+  for key, value in grafeas_occurrence.intoto["signed"].byproducts["custom_values"].items():
+    if key == "return-value":
+      byproducts[key] = int(value)  # cast return-value back to int
+    else:
+      byproducts[key] = value
+
+  for key, value in grafeas_occurrence.intoto["signed"].environment.items():
+    if key == "custom_values":
+      continue
+    environment[key] = value
+
+  for key, value in grafeas_occurrence.intoto["signed"].environment["custom_values"].items():
+    environment[key] = value
+
+  in_toto_link = Link(name=step_name, materials=materials, products=products, byproducts=byproducts, command=command, environment=environment)
+
+  signatures = []
+  for signature in grafeas_occurrence.intoto["signatures"]:
+    if "sig" not in signature:
+      signatures.append({"keyid": signature["keyid"], "sig": signature["signature"]})
+    else:
+      signatures.append(signature)
+
+  return Metablock(signed=in_toto_link, signatures=signatures)
+
+
+class GrafeasInTotoTransport:
+  server_url = None
+
+  def __init__(self, server_url):
+    self.server_url = server_url
+
+  def dispatch(self, in_toto_link, note_name, resource_uri):
```

Comment:
This is following the signature from the jenkins transports right? Is this a conscious decision? I wonder if instead we'd like to have something that follows the CRUD api nomenclature from the Grafeas server:

```
transport = GrafeasInTotoTransport("https://nyu.wtf")
transport.create(my_link)
my_link = transport.read(name)
transport.update(name, my_link)
transport.delete(name)
```